frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import {
+  accessModeRadios,
+  volumeModeRadios,
+  initialAccessModes,
+  provisionerAccessModeMapping,
+  dropdownUnits,
+  getAccessModeForProvisioner,
+} from '@console/internal/components/storage/shared';
+import { RadioInput } from '@console/internal/components//radio';
+import { RequestSizeInput, ExpandCollapse } from '@console/internal/components/utils';
+
+import { isCephProvisioner, cephStorageProvisioners } from '@console/shared/src/utils';
+
+interface VolumeClaimTemplateFormProps {
+  name: string;
+  initialSizeValue?: string;
+  initialSizeUnit?: string;
+  initialVolumeMode?: string;
+}
+
+const VolumeClaimTemplateForm: React.FC<VolumeClaimTemplateFormProps> = ({
+  name,
+  initialSizeValue = '1',
+  initialSizeUnit = 'Gi',
+  initialVolumeMode = 'Filesystem',
+}) => {
+  const [accessModeHelp, setAccessModeHelp] = React.useState('Permissions to the mounted drive.');
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const [allowedAccessModes, setAllowedAccessModes] = React.useState(initialAccessModes);
+  const [volumeMode, setVolumeMode] = React.useState(initialVolumeMode);
+  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeValue, setRequestSizeValue] = React.useState(initialSizeValue);
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState(initialSizeUnit);
+  const [storageProvisioner, setStorageProvisioner] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState(initialSizeUnit);
+
+  const { t } = useTranslation();
+
+  const handleAccessMode: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setAccessMode(event.currentTarget.value);
+  };
+
+  const handleStorageClass = (updatedStorageClass) => {
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? 'Access mode is set by storage class and cannot be changed'
+        : 'Permissions to the mounted drive';
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
```

Comment:
This feels weird... I wonder if we need some interim handler or wrapper of the StorageClass to simplify this logic. It's a dropdown after all.

Reply:
The same login applies for the CreatePVC dropdown. Hence, kept it. I dont think we can move it out as a util as it is directly changing the states of this component.

However, should we do this validation on client or should we let the server take care of this?

Reply:
I'll have to relook at this once you clean things up (other comments). It's possible there is a way to streamline this so it's not so heavy.

Reply:
Still need to check on this, but it should be fine as-is if I cannot figure out a better solution.